add support for debian bullseye

duviful opened the pull request:
Add support for Debian bullseye, which doesn't have python3.8 (only 3.9).
Bumped ansible-role-postgresql version in molecule if we want at some point add debian in tests. This version comes with support for Debian-11.

Feel free to modify the code to adapt to your coding style.

jvoss commented on tasks/variables.yml:
```diff
@@ -0,0 +1,10 @@
+  ansible.builtin.include_vars: "{{ ansible_distribution }}-{{ ansible_distribution_version.split('.')[0] }}.yml"
+  when: ansible_os_family == 'Debian'
+
+- name: Define python_packages.
```
This task winds up being triggered when not Debian (Ubuntu and RHEL) as well and failing.

I like the method you are using here however either all supported distros will need refactored to this style or you will need to add a conditional here for Debian as well.

I am also happy if you add a conditional here for now and want me to refactor the rest.

jvoss replied:
Thank you very much for this!

jvoss commented on tasks/variables.yml:
```diff
@@ -0,0 +1,10 @@
+- name: Define python_packages.
+  ansible.builtin.set_fact:
+    python_packages: "{{ __python_packages | list }}"
+  when: python_packages is not defined
```
Temporary workaround (see note above):
```
when:
  - ansible_os_family == 'Debian'
  - python_packages is not defined

jvoss replied:
Thanks for adding the other distro variables!

jvoss reviewed (changes requested):
Thank you for adding Debian support! I like the progress however there is one item to address that is breaking installation on other distributions.

duviful commented:
I'm also adding some more variables to have `tasks/upgrade.yml` properly working with the changes I've introduced.

duviful commented:
my local molecule fails at

```
TASK [geerlingguy.postgresql : Ensure PostgreSQL is started and enabled on boot.] ***
fatal: [instance]: FAILED! => {"changed": false, "cmd": "/bin/systemctl", "msg": "Failed to connect to bus: No such file or directory", "rc": 1, "stderr": "Failed to connect to bus: No such file or directory\n", "stderr_lines": ["Failed to connect to bus: No such file or directory"], "stdout": "", "stdout_lines": []}
```

do you use it for test or you moved to github actions fro testing?

jvoss commented:
> my local molecule fails at
> 
> ```
> TASK [geerlingguy.postgresql : Ensure PostgreSQL is started and enabled on boot.] ***
> fatal: [instance]: FAILED! => {"changed": false, "cmd": "/bin/systemctl", "msg": "Failed to connect to bus: No such file or directory", "rc": 1, "stderr": "Failed to connect to bus: No such file or directory\n", "stderr_lines": ["Failed to connect to bus: No such file or directory"], "stdout": "", "stdout_lines": []}
> ```
> 
> do you use it for test or you moved to github actions fro testing?

I also have had a lot of problems with `geerlingguy.postgresql` for local testing as well, at least on OSX. I'm curious if developing on Linux has the same issue considering the GitHub actions run fine.

I have been relying more on testing during the PR checks or manually with a test VM but I too wish I could rely on local molecule testing.

jvoss reviewed (changes requested):
I think we are getting closer. There is a potential order of operations issue that is causing some of the builds to fail (see review comments).

Feel free to add the appropriate Debian distros the GitHub actions matrices to include in testing if you like. Otherwise I will be adding them at the completion of this PR to ensure we do not break it in the future.

duviful commented on vars/Amazon-2.yml:
```diff
@@ -0,0 +1,3 @@
+---
+__python_version: "3.8"
+__python_packages: "python{{ python_version }}"
```
I think you are right and I like the solution you provided.

jvoss commented:
> I'm sorry for this trial and error approach but without proper testing, i' just trying...

It's no problem at all! This is what PRs are for. Feel free to send as many commits as you need.

duviful commented:
I moved my local molecule testing on a amd64 machine and I'm using molecule-vagrant. 
I added boxes for all the OS flavors.
I still got some non-zero return codes during configure.py to sort out.

For example, debian 10 (on python 3.7) gets a warning from netbox 3.11 that will not be supported in the future. this is interpreted as a non zero by ansible and fails the task.
